golos.publication/golos.publication.hpp

```diff
@@ -14,23 +14,22 @@ class publication : public contract {
     void set_rules(const funcparams& mainfunc, const funcparams& curationfunc, const funcparams& timepenalty,
         int64_t curatorsprop, int64_t maxtokenprop, symbol tokensymbol);
     void on_transfer(name from, name to, asset quantity, std::string memo);
-    void create_message(name account, std::string permlink, name parentacc, std::string parentprmlnk,
+    void create_message(structures::mssgid message_id, structures::mssgid parent_id,
         std::vector<structures::beneficiary> beneficiaries, int64_t tokenprop, bool vestpayment,
         std::string headermssg, std::string bodymssg, std::string languagemssg, std::vector<structures::tag> tags,
         std::string jsonmetadata);
-    void update_message(name account, std::string permlink,
-        std::string headermssg, std::string bodymssg, std::string languagemssg, std::vector<structures::tag> tags,
-        std::string jsonmetadata);
-    void delete_message(name account, std::string permlink);
-    void upvote(name voter, name author, std::string permlink, uint16_t weight);
-    void downvote(name voter, name author, std::string permlink, uint16_t weight);
-    void unvote(name voter, name author, std::string permlink);
-    void close_message(name account, uint64_t id);
+    void update_message(structures::mssgid message_id, std::string headermssg, std::string bodymssg,
+                        std::string languagemssg, std::vector<structures::tag> tags, std::string jsonmetadata);
+    void delete_message(structures::mssgid message_id);
+    void upvote(name voter, structures::mssgid message_id, uint16_t weight);
+    void downvote(name voter, structures::mssgid message_id, uint16_t weight);
+    void unvote(name voter, structures::mssgid message_id);
+    void close_message(structures::mssgid message_id);
     void set_params(std::vector<posting_params> params);
-    void reblog(name rebloger, name author, std::string permlink);
+    void reblog(name rebloger, structures::mssgid message_id);
 private:
-    void close_message_timer(name account, uint64_t id, uint64_t delay_sec);
-    void set_vote(name voter, name author, std::string permlink, int16_t weight);
+    void close_message_timer(structures::mssgid message_id, uint64_t id, uint64_t delay_sec);
+    void set_vote(name voter, name author, std::string permlink, uint64_t ref_block_num, int16_t weight);
```

Comment:
It's better to pass `author,permlink,ref_block_num` as mssgid to this function too.